sdk/scheduler/src/test/java/com/mesosphere/sdk/framework/SchedulerDriverFactoryTest.java

```diff
@@ -125,25 +134,59 @@ public void testMissingPrincipalNoAuth() throws Exception {
     }
 
     @Test(expected=IllegalArgumentException.class)
-    public void testMissingPrincipalSidechannel() throws Exception {
+    public void testMissingPrincipalSidechannel() {
         when(mockSchedulerConfig.isSideChannelActive()).thenReturn(true);
         new CustomSchedulerDriverFactory().create(
                 new NoOpScheduler(), FRAMEWORK_WITHOUT_PRINCIPAL, MASTER_URL, mockSchedulerConfig);
     }
 
     @Test(expected=IllegalArgumentException.class)
-    public void testMissingPrincipalWithSecret() throws Exception {
+    public void testMissingPrincipalWithSecret() {
         new CustomSchedulerDriverFactory().create(
                 new NoOpScheduler(), FRAMEWORK_WITHOUT_PRINCIPAL, MASTER_URL, mockSchedulerConfig, SECRET);
     }
 
     @Test(expected=IllegalArgumentException.class)
-    public void testMissingPrincipalWithSecretAndSidechannel() throws Exception {
+    public void testMissingPrincipalWithSecretAndSidechannel() {
         when(mockSchedulerConfig.isSideChannelActive()).thenReturn(true);
         new CustomSchedulerDriverFactory().create(
                 new NoOpScheduler(), FRAMEWORK_WITHOUT_PRINCIPAL, MASTER_URL, mockSchedulerConfig, SECRET);
     }
 
+    @Test
+    public void createInternalTest() {
+        // This should have been an integration test but upstream has no version information (and it shouldn't have ?)
+        Random r = new Random();
+        for (ImmutableTriple<Boolean, String, ? extends Class<? extends Mesos>> data : Arrays.asList(
+                // Triple of (Boolean V1 Capability Supported | Requested Version | Created Version)
+                ImmutableTriple.of(true, SchedulerConfig.MESOS_API_VERSION_V1, V1Mesos.class),
+                ImmutableTriple.of(r.nextBoolean(), "NOT_V1", V0Mesos.class)
+        )) {
+            Capabilities capabilities = mock(Capabilities.class);
+            when(capabilities.supportsV1APIByDefault()).thenReturn(data.left);
+            try {
+                assertTrue(
+                        new SchedulerDriverFactory().startInternalCustom(
+                                capabilities,
+                                mock(FrameworkInfo.class),
+                                "masterUrl",
+                                mock(Credential.class),
+                                data.middle
+                        )
+                        .apply(mock(MesosToSchedulerDriverAdapter.class))
+                        .getClass()
```

Comment:
Could this be done as `assertTrue([...].getClass() instanceof data.right)`? Or is that the issue?

Reply:
I am getting a compile time error if I do a `instanceof` as `data.right` is of type `? extends Class<? extends Mesos>`